pandas/tests/io/test_pickle.py

```diff
@@ -121,47 +105,19 @@ def compare_series_ts(result, expected, typ, version):
 
 
 def compare_series_dt_tz(result, expected, typ, version):
-    # 8260
-    # dtype is object < 0.17.0
-    if LooseVersion(version) < LooseVersion('0.17.0'):
-        expected = expected.astype(object)
-        tm.assert_series_equal(result, expected)
-    else:
-        tm.assert_series_equal(result, expected)
+    tm.assert_series_equal(result, expected)
 
 
 def compare_series_cat(result, expected, typ, version):
-    # Categorical dtype is added in 0.15.0
-    # ordered is changed in 0.16.0
-    if LooseVersion(version) < LooseVersion('0.15.0'):
-        tm.assert_series_equal(result, expected, check_dtype=False,
-                               check_categorical=False)
-    elif LooseVersion(version) < LooseVersion('0.16.0'):
-        tm.assert_series_equal(result, expected, check_categorical=False)
-    else:
-        tm.assert_series_equal(result, expected)
+    tm.assert_series_equal(result, expected)
```

Comment:
Could maybe simplify these methods since they don't have combat checks any more but I think better left as a separate exercise (happy to open issue)

Reply:
sgtm.